File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      question?(input) -> "Sure."
      nothing?(input)  -> "Fine. Be that way!"
      yelling?(input)  -> "Whoa, chill out!"
      true             -> "Whatever."
    end
  end

  defp question?(input), do: String.ends_with?(input, "?")
  defp nothing?(input), do: String.trim(input) == ""
  defp yelling?(input) do
    # String is all upper-case and has at least one letter.
    String.upcase(input) == input && String.match?(input, ~r/\p{L}/u)
  end
end

File: go/bob/bob.go
package bob // package name must match the package name in bob_test.go

import (
	"regexp"
	"strings"
)

const testVersion = 2 // same as targetTestVersion

func Hey(i string) string {
	i = strings.TrimSpace(i)
	if isNothing(i) {
		return "Fine. Be that way!"
	}
	if isYelling(i) {
		return "Whoa, chill out!"
	}
	if isQuestion(i) {
		return "Sure."
	}
	return "Whatever."
}

func isNothing(i string) bool {
	return len(i) == 0
}

func isYelling(i string) bool {
	isAllUpper := strings.ToUpper(i) == i
	r := regexp.MustCompile("[A-z]")
	hasLetters := r.MatchString(i)

	return isAllUpper && hasLetters
}

func isQuestion(i string) bool {
	return strings.HasSuffix(i, "?")
}
